File: Perk.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__New = ____lualib.__TS__New
local ____exports = {}
local ____Enums = require("Enums")
local Perks = ____Enums.Perks
local ____Helpers = require("Helpers")
local exhaustiveGuard = ____Helpers.exhaustiveGuard
____exports.default = __TS__Class()
local Perk = ____exports.default
Perk.name = "Perk"
function Perk.prototype.____constructor(self, gameManager, perkType)
    self.gameManager = gameManager
    self.perkType = perkType
end
function Perk.prototype.save(self)
    return {perkType = self.perkType}
end
function Perk.load(self, gameManager, data)
    return __TS__New(____exports.default, gameManager, data.perkType)
end
function Perk.prototype.getPerkName(self)
    repeat
        local ____switch6 = self.perkType
        local ____cond6 = ____switch6 == Perks.EXTRA_CARD
        if ____cond6 then
            return "Extra Card"
        end
        ____cond6 = ____cond6 or ____switch6 == Perks.EXTRA_DISCARD
        if ____cond6 then
            return "Extra Discard"
        end
        ____cond6 = ____cond6 or ____switch6 == Perks.INCREASED_LOOT
        if ____cond6 then
            return "Increased Loot"
        end
        do
            exhaustiveGuard(self.perkType)
        end
    until true
end
return ____exports

File: Perk.ts
/** @noSelfInFile */

import { Perks } from "Enums"
import GameManager from "GameManager"
import { exhaustiveGuard } from "Helpers"

export interface PerkData {
    perkType: Perks
}

export default class Perk {
    gameManager: GameManager
    perkType: Perks

    constructor(gameManager: GameManager, perkType: Perks) {
        this.gameManager = gameManager
        this.perkType = perkType
    }

    save(): PerkData {
        return {
            perkType: this.perkType
        }
    }

    static load(gameManager: GameManager, data: PerkData): Perk {
        return new Perk(gameManager, data.perkType)
    }

    getPerkName(): string {
        switch (this.perkType) {
            case Perks.EXTRA_CARD:
                return "Extra Card"
            case Perks.EXTRA_DISCARD:
                return "Extra Discard"
            case Perks.INCREASED_LOOT:
                return "Increased Loot"
            default:
                exhaustiveGuard(this.perkType)
        }
    }
}
